File: 1630. Arithmetic Subarrays/Solution.cpp
class Solution {
public:
    vector<bool> checkArithmeticSubarrays(vector<int>& nums, vector<int>& l, vector<int>& r) {
        vector<bool> vRet(l.size(), true);
        
        for(int i=0;i<l.size();i++)
        {
            vector<int> vTmp = vector<int>(nums.begin() + l[i], nums.begin() + r[i] + 1);
            std::sort(vTmp.begin(), vTmp.end());
            
            int diff = vTmp[0] - vTmp[1];
            for(int j=0;j<vTmp.size() - 1;j++)
            {
                if((vTmp[j] - vTmp[j+1]) != diff)
                {
                    vRet[i] = false;
                    break;
                }
            }
        }
        return vRet;
    }
};

File: 1630. Arithmetic Subarrays/Solution.py
class Solution:
    def checkArithmeticSubarrays(self, nums: List[int], l: List[int], r: List[int]) -> List[bool]:
        lBool = [True] * len(l)
        
        for i in range(len(l)):
            lTmp = nums[l[i]:r[i]+1]
            lTmp.sort()
            iDiff = lTmp[0] - lTmp[1]
            for j in range(len(lTmp) - 1):
                if lTmp[j] - lTmp[j+1] != iDiff:
                    lBool[i] = False
        
        return lBool
        
